test/views/pages/ip2016/SummaryViewSpec.scala

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.pages.ip2016
+
+import enums.ApplicationType
+import models.{SummaryModel, SummaryRowModel, SummarySectionModel}
+import org.jsoup.Jsoup
+import play.api.i18n.Messages.Implicits._
+import testHelpers.ViewSpecHelpers.CommonViewSpecHelper
+import testHelpers.ViewSpecHelpers.IP16.SummaryViewMessages
+import utils.CallMap
+import views.html.pages.ip2016.{summary => views}
+
+class SummaryViewSpec  extends CommonViewSpecHelper with SummaryViewMessages{
+  def totalPensionsAmountSummaryRow(totalAmount: String) = SummaryRowModel("totalPensionsAmt", None, None, true, totalAmount)
+
+  val psoDetailsSummaryRow = SummaryRowModel("psoDetails", Some(controllers.routes.IP2016Controller.psoDetails), Some(controllers.routes.IP2016Controller.removePsoDetails), false, "£10,000", "1 February 2016")
+
+  "the SummaryView" should{
```

Comment:
This test requires that the dynamic nature of the display is tested. Please show that the values are different based on the different data input with another variant of the elements with different setup data.